python-sdk/nuscenes/eval/detection/data_classes.py

```diff
@@ -290,36 +308,49 @@ def mean_ap(self):
         return np.mean([np.mean(aps) for aps in self.label_aps.values()])
 
     @property
-    def tp_scores(self):
-        """ Calculates the mean true positive score across all classes for each metric. """
-        tp_scores = {}
+    def tp_errors(self):
+        """ Calculates the mean true positive error across all classes for each metric. """
+        tp_errors = {}
         for metric_name in TP_METRICS:
-            scores = []
+            errors = []
             for detection_name in self.cfg.class_names:
-                if detection_name in ['barrier', 'traffic_cone'] and metric_name == 'attr_err':
-                    continue  # There are no attributes for these classes, so don't count them.
+                if detection_name in ['traffic_cone'] and metric_name in ['attr_err', 'vel_err', 'orient_err']:
+                    # We don't include this in mean since:
+                    # - We dont have attributes for cones.
+                    # - Cones are stationary.
+                    # - Orientation of a cone is ill-defined.
+                    continue
 
-                # We convert the true positive errors to "scores" by 1-error
-                score = 1.0 - self.label_tp_errors[detection_name][metric_name]
+                if detection_name in ['barrier'] and metric_name in ['attr_err', 'vel_err']:
+                    # We don't include this in mean since:
+                    # - We dont have attributes for cones.
+                    # - Barriers are stationary.
+                    continue
 
-                # Some of the true positive errors are unbounded, so we bound the scores to min 0.
-                score = max(0.0, score)
+                errors.append(self.label_tp_errors[detection_name][metric_name])
 
-                scores.append(score)
-            tp_scores[metric_name] = np.mean(scores)
-        return tp_scores
+            tp_errors[metric_name] = float(np.mean(errors))
+        return tp_errors
 
     @property
     def weighted_sum(self):
         weighted_sum = self.cfg.mean_ap_weight * self.mean_ap
         for metric_name in TP_METRICS:
-            weighted_sum += self.tp_scores[metric_name]
+
+            # We convert the true positive errors to "scores" by 1-error
+            score = 1.0 - self.tp_errors[metric_name]
+
+            # Some of the true positive errors are unbounded, so we bound the scores to min 0.
+            score = max(0.0, score)
+
+            # Accumulate
+            weighted_sum += score
         return weighted_sum / float(self.cfg.mean_ap_weight + len(TP_METRICS))
 
     def serialize(self):
         return {'label_aps': self.label_aps,
-                'label_tp_errors': self.label_tp_errors,
+                'label_tp_metrics': self.label_tp_errors,
```

Comment:
This reverts the dictionary keys back to the original template. But for the computation I thought it was more clear to call it `errors` to separate it out from `scores` later on.

Reply:
I'd suggest to change metrics to errors everywhere.

Reply:
Great. So you OK changing the names in the final output `json`? If so I might tweak one or more things while I am at it ;)

Reply:
Let's go nuts on it.